src/components/plans/FixedChargeAccordion.tsx

```diff
@@ -0,0 +1,359 @@
+        isPayInAdvance: formikCharge.payInAdvance || false,
+        chargeModel: formikCharge.chargeModel,
+      })
+
+      return {
+        chargeModelComboboxData: localChargeModelComboboxData,
+        hasDefaultPropertiesErrors:
+          typeof fixedChargeErrors === 'object' &&
+          typeof fixedChargeErrors[index] === 'object' &&
+          typeof fixedChargeErrors[index].properties === 'object',
+        initialLocalCharge: formikProps.initialValues.fixedCharges[index],
+        isPayInAdvanceOptionDisabled: localIsPayInAdvanceOptionDisabled,
+        isProratedOptionDisabled: localIsProratedOptionDisabled,
+        localCharge: formikCharge,
+      }
+    }, [
+      fixedChargeErrors,
+      formikProps.initialValues.fixedCharges,
+      formikProps.values.fixedCharges,
+      getFixedChargeModelComboboxData,
+      getIsPayInAdvanceOptionDisabledForFixedCharge,
+      getIsProRatedOptionDisabledForFixedCharge,
+      index,
+    ])
+
+    const handleUpdate = useCallback(
+      (
+        name: HandleUpdateFixedChargesProps['name'],
+        value: HandleUpdateFixedChargesProps['value'],
+      ) => {
+        handleUpdateFixedCharges({
+          setFieldValue: formikProps.setFieldValue,
+          index,
+          localCharge,
+          name,
+          value,
+        })
+      },
+
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      [index, localCharge],
+    )
+
+    const taxValueForBadgeDisplay = useMemo((): string | undefined => {
+      if (!localCharge?.taxes?.length && !formikProps?.values?.taxes?.length) return
+
+      if (localCharge.taxes?.length)
+        return String(localCharge.taxes.reduce((acc, cur) => acc + cur.rate, 0))
+
+      return String(formikProps?.values?.taxes?.reduce((acc, cur) => acc + cur.rate, 0))
+    }, [formikProps?.values?.taxes, localCharge.taxes])
+
+    // Fixed charges don't have a pricing unit
+    const chargePricingUnitShortName = undefined
+
+    const isAnnual = [PlanInterval.Semiannual, PlanInterval.Yearly].includes(
+      formikProps.values.interval,
+    )
+
+    return (
+      <Accordion
+        noContentMargin
+        id={id}
+        initiallyOpen={!!(isInitiallyOpen || !formikProps.values.fixedCharges?.[index]?.id)}
+        summary={
+          <div className="flex w-full items-center justify-between gap-3 overflow-hidden">
+            <div className="flex flex-col overflow-hidden">
+              <div className="flex items-center gap-2 truncate p-1 pl-0">
+                <Typography variant="bodyHl" color="textSecondary" noWrap>
+                  {localCharge.invoiceDisplayName || localCharge?.addOn?.name}
+                </Typography>
+                <EditInvoiceDisplayNameButton
+                  editInvoiceDisplayNameDialogRef={editInvoiceDisplayNameDialogRef}
+                  currentInvoiceDisplayName={localCharge.invoiceDisplayName}
+                  onEdit={(invoiceDisplayName: string) => {
```

Comment:
Extract in a method so our component body is as clean as possible

Reply:
Done